nixos/modules/services/misc/matrix-synapse.nix

```diff
@@ -94,46 +125,97 @@ in {
           PEM dh parameters for ephemeral keys
         '';
       };
-      bind_port = mkOption {
-        type = types.int;
-        default = 8448;
+      server_name = mkOption {
+        type = types.str;
+        example = "example.com";
         description = ''
-          The port to listen for HTTPS requests on.
-          For when matrix traffic is sent directly to synapse.
+          The domain name of the server, with optional explicit port.
+          This is used by remote servers to connect to this server,
+          e.g. matrix.org, localhost:8080, etc.
+          This is also the last part of your UserID.
         '';
       };
-      unsecure_port = mkOption {
-        type = types.int;
-        default = 8008;
+      web_client = mkOption {
+        type = types.bool;
+        default = false;
         description = ''
-          The port to listen for HTTP requests on.
-          For when matrix traffic passes through loadbalancer that unwraps TLS.
+          Whether to serve a web client from the HTTP/HTTPS root resource.
         '';
       };
-      bind_host = mkOption {
+      public_baseurl = mkOption {
         type = types.str;
         default = "";
+        example = "https://example.com:8448/";
         description = ''
-          Local interface to listen on.
-          The empty string will cause synapse to listen on all interfaces.
+          The public-facing base URL for the client API (not including _matrix/...)
         '';
       };
-      server_name = mkOption {
+      listeners = mkOption {
+        type = types.listOf types.attrs;
+        default = [{
+          port = 8448;
```

Comment:
port has to be a string here, otherwise you get an error :

`cannot coerce an integer to a string, at <nixpkgs>/nixos/modules/services/misc/matrix-synapse.nix:10:19`